File: bin/spark/application/accounts-by-state_project/src/main/scala/solution/AccountsByState.scala
package solution

import org.apache.spark.sql.SparkSession

object AccountsByState {
  def main(args: Array[String]) {
    if (args.length < 1) {
      System.err.println("Usage: solution.AccountsByState <state-code>")
      System.exit(1)
    }
 
    val stateCode = args(0)
    
    val spark = SparkSession.builder.getOrCreate()
    spark.sparkContext.setLogLevel("WARN")

    val accountsDF = spark.read.table("devsh.accounts")
    val stateAccountsDF = accountsDF.where(accountsDF("state") === stateCode)
    stateAccountsDF.write.mode("overwrite").save("/devsh_loudacre/accounts_by_state/"+stateCode)

    spark.stop
  }
}

File: bin/spark/application/python-bonus/accounts-by-state.py
# Python application bonus exercise solution
import sys

from pyspark.sql import SparkSession

if __name__ == "__main__":
  if len(sys.argv) < 2:
    print >> sys.stderr, "Usage: accounts-by-state.py <state-code>"
    sys.exit()

  print "len=",len(sys.argv)
  stateCode = sys.argv[1]

  spark = SparkSession.builder.appName("Accounts by State 3").getOrCreate()
  spark.sparkContext.setLogLevel("WARN")
  
  accountsDF = spark.read.table("devsh.accounts")
  stateAccountsDF = accountsDF.where(accountsDF.state == stateCode)
  stateAccountsDF.write.mode("overwrite").save("/devsh_loudacre/accounts_by_state/" + stateCode)

  spark.stop()